### ISSUE TITLE
Create a simple client

### ISSUE THREAD

**Issue by droidmonkey** [nitroshare/qhttpengine#22]
It would be great to have a simple HTTP client that is the compliment to the Server. The client would take in a QUrl and create the QTcpSocket and QHttpEngine::Socket to communicate with the server. The client would also automatically handle SSL connections and redirects (with a default maximum).

**Comment by droidmonkey** [comment]
Note: this will actually require modifying Socket such that it acts as a Request **or** Response Socket. Right  now Socket only parses the Request headers as if it was acting for a Server.

**Comment by nathan-osman** [comment]
Yes, and I would probably consider splitting the class up at that point. There would be a base class `Socket` and two classes that derive from it, `ServerSocket` and `ClientSocket`. That way the common functionality could remain where it is and the parts of the protocol specific to each end would live in their respective classes.

**Comment by droidmonkey** [comment]
If you don't mind I'm going to open several minor issues and suggestions as I learn more about your code. I'd be willing to help fix them as well.

**Comment by nathan-osman** [comment]
Awesome, that sounds great.
